tests/unit/integrations/academy-integration-test.php

```diff
@@ -0,0 +1,290 @@
+<?php
+
+namespace Yoast\WP\SEO\Tests\Unit\Integrations;
+
+use Brain\Monkey;
+use Mockery;
+use WPSEO_Admin_Asset_Manager;
+use Yoast\WP\SEO\Conditionals\Admin_Conditional;
+use Yoast\WP\SEO\Conditionals\User_Can_Manage_Wpseo_Options_Conditional;
+use Yoast\WP\SEO\Helpers\Current_Page_Helper;
+use Yoast\WP\SEO\Helpers\Product_Helper;
+use Yoast\WP\SEO\Integrations\Academy_Integration;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+use Yoast\WP\SEO\Helpers\Short_Link_Helper;
+
+/**
+ * Class Academy_Integration_Test.
+ *
+ * @coversDefaultClass \Yoast\WP\SEO\Integrations\Academy_Integration
+ */
+class Academy_Integration_Test extends TestCase {
+
+	const PAGE = 'wpseo_page_academy';
+
+	/**
+	 * Holds the WPSEO_Admin_Asset_Manager.
+	 *
+	 * @var WPSEO_Admin_Asset_Manager
+	 */
+	private $asset_manager;
+
+	/**
+	 * Holds the Current_Page_Helper.
+	 *
+	 * @var Current_Page_Helper
+	 */
+	private $current_page_helper;
+
+	/**
+	 * Holds the Product_Helper.
+	 *
+	 * @var Product_Helper
+	 */
+	private $product_helper;
+
+	/**
+	 * Holds the Short_Link_Helper.
+	 *
+	 * @var Short_Link_Helper
+	 */
+	private $shortlink_helper;
+
+	/**
+	 * The class under test.
+	 *
+	 * @var Academy_Integration
+	 */
+	protected $instance;
+
+	/**
+	 * Runs the setup to prepare the needed instance
+	 */
+	public function set_up() {
+		$this->stubTranslationFunctions();
+
+		$this->asset_manager       = Mockery::mock( WPSEO_Admin_Asset_Manager::class );
+		$this->current_page_helper = Mockery::mock( Current_Page_Helper::class );
+		$this->product_helper      = Mockery::mock( Product_Helper::class );
+		$this->shortlink_helper    = Mockery::mock( Short_Link_Helper::class );
+
+		$this->instance = new Academy_Integration(
+			$this->asset_manager,
+			$this->current_page_helper,
+			$this->product_helper,
+			$this->shortlink_helper
+		);
+	}
+
+	/**
+	 * Tests the retrieval of the conditionals.
+	 *
+	 * @covers ::get_conditionals
+	 */
+	public function test_get_conditionals() {
+		static::assertEquals(
+			[
+				Admin_Conditional::class,
+				User_Can_Manage_Wpseo_Options_Conditional::class,
+			],
+			Academy_Integration::get_conditionals()
+		);
+	}
+
+	/**
+	 * Provider for test_register_hooks
+	 *
+	 * @return array
+	 */
+	public function register_hooks_provider() {
+		return [
+			'Not on academy page' => [
+				'current_page'          => 'not academy page',
+				'times for each action' => 0,
```

Comment:
Interesting, I thought the key needed to be the same as the param ($action_times in this case)